Objects/typeobject.c

```diff
@@ -9045,6 +9045,26 @@ super_init(PyObject *self, PyObject *args, PyObject *kwds)
             return -1;
         Py_INCREF(obj);
     }
+
+    if (PyType_Check(type) && PyObject_IsSubclass((PyObject*)type, PyExc_BaseException) && PyObject_IsInstance(obj, (PyObject*)type)) {
```

Comment:
PyObject_IsSubclass() can throw an error, in which case it returns -1. This needs to be checked. Many other calls below are also missing error checks.

More broadly it seems wrong to have BaseException-specific code here, and it is fragile to check the flags of the code object directly here. I think you need to go back and come up with a more robust design for this feature.